pkg/skaffold/initializer/init.go

```diff
@@ -249,16 +249,53 @@ func detectBuilders(enableJibInit bool, path string) ([]InitBuilder, error) {
 func processCliArtifacts(artifacts []string) ([]builderImagePair, error) {
 	var pairs []builderImagePair
 	for _, artifact := range artifacts {
-		parts := strings.Split(artifact, "=")
-		if len(parts) != 2 {
-			return nil, fmt.Errorf("malformed artifact provided: %s", artifact)
+		// Parses JSON in the form of: {"builder":"Name of Builder","payload":{...},"image":"image.name"}.
+		// The builder field is parsed first to determine the builder type, and the payload is parsed
+		// afterwards once the type is determined.
+		a := struct {
+			Name  string `json:"builder"`
+			Image string `json:"image"`
+		}{}
+		if err := json.Unmarshal([]byte(artifact), &a); err != nil {
+			parts := strings.Split(artifact, "=")
+			if len(parts) != 2 {
+				return nil, fmt.Errorf("malformed artifact provided: %s", artifact)
+			}
+
+			// Not JSON, use backwards compatible method
```

Comment:
nit: move up the comment right after `Unmarshal`.

Reply:
Oops, forgot to commit this.